libs/designsystem/src/lib/directives/floating/floating.directive.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  Directive,
+  ElementRef,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Renderer2,
+} from '@angular/core';
+import { autoPlacement, autoUpdate, computePosition, offset, shift } from '@floating-ui/dom';
+import { ComputePositionConfig, Middleware, Placement } from '@floating-ui/core/src/types';
+import { DesignTokenHelper } from '@kirbydesign/core';
+
+export type TriggerEvent = 'hover' | 'click' | 'focus';
+
+export enum FloatingOffset {
+  none = 0,
+  small = 5,
+  medium = 10,
+  large = 15,
+}
+
+interface EventMethods {
+  event: string;
+  method: () => void;
+}
+
+@Directive({
+  selector: '[kirbyFloating]',
+  standalone: true,
+})
+export class FloatingDirective implements OnInit, OnDestroy {
+  @Input() private set reference(ref: ElementRef) {
+    this.tearDownEventHandling();
+    this._reference = ref;
+    this.setupEventHandling();
+  }
+
+  private get reference(): ElementRef {
+    return this._reference;
+  }
+
+  @Input() private set placement(placement: Placement) {
+    this._placement = placement;
+    this.updateHostElementPosition();
+  }
+
+  private get placement() {
+    return this._placement;
+  }
+
+  @Input() private set triggers(eventTriggers: Array<TriggerEvent> | TriggerEvent) {
+    this._triggers = eventTriggers;
+    this.tearDownEventHandling();
+    this.setupEventHandling();
+  }
+
+  private get triggers(): Array<TriggerEvent> | TriggerEvent {
+    return this._triggers;
+  }
+
+  @Input() private isDisabled: boolean = false;
+
+  // Displaces the floating element from its core placement along the specified axes.
+  @Input() private offset: FloatingOffset = FloatingOffset.none;
+
+  // Moves the floating element along the specified axes in order to keep it in view.
+  // This does not always work as expected, so don't "just" set it.
+  @Input() private shift: boolean = false;
+
+  // Chooses the placement that has the most space available automatically.
+  @Input() private autoPlacement: boolean = false;
+
+  @Input() private closeOnSelect: boolean = true;
+  @Input() private closeOnEscapeKey: boolean = true;
+  @Input() private closeOnBackdrop: boolean = true;
+
+  @HostListener('document:keydown.escape', ['$event']) public onEscapeKeyPressed() {
+    if (this.closeOnEscapeKey) {
+      this.hide();
+    }
+  }
+
+  @HostListener('document:mousedown', ['$event'])
+  public onMouseClick(event): void {
+    const clickedOnHost: boolean = this.elementRef.nativeElement.contains(event.target);
+    clickedOnHost ? this.handleClickInsideHostElement() : this.handleClickOutsideHostElement(event);
+  }
+
+  private _placement: Placement;
+  private _triggers: Array<TriggerEvent> | TriggerEvent = 'click';
```

Comment:
I suspect that 80% of the cases, the user would need only one trigger. It would then be kind of cumbersome to always have to put in an array, due to still supporting the 20% of the cases where we have two or more triggers.

As I see it, it's a QoL for the user, while the maintainer gets a bit of extra work. 

I'm not married to one or the either and can see pros/cons on both, so if you have a preference we can go with that.

Reply:
How about a nice compromise and just always convert it to an array in the setter, so we always know what we are working with.